iquip/apps/builder.py

```diff
@@ -107,30 +295,47 @@ def __init__(
         """Extended.
         
         Args:
-            experimentPath: The path of the experiment file.
-            experimentClsName: The class name of the experiment.
+            experimentPath, experimentClsName: See the attributes section in BuilderApp.
             experimentInfo: The experiment information, a dictionary of protocols.ExperimentInfo.
         """
         super().__init__(name, parent=parent)
         self.experimentPath = experimentPath
         self.experimentClsName = experimentClsName
-        self.experimentInfo = ExperimentInfo(**experimentInfo)
         self.builderFrame = BuilderFrame()
+        self.initArgsEntry(ExperimentInfo(**experimentInfo))
         # connect signals to slots
         self.builderFrame.submitButton.clicked.connect(self.submit)
 
+    def initArgsEntry(self, experimentInfo: ExperimentInfo):
+        """Initializes the build arguments entry.
+        
+        Args:
+            experimentInfo: The experiment information.
+        """
+        for argName, (argInfo, *_) in experimentInfo.arginfo.items():
+            entryCls = {
+                "BooleanValue": _BooleanEntry,
+                "StringValue": _StringEntry,
+                "EnumerationValue": _EnumerationEntry,
+                "NumberValue": _NumberEntry
+            }[argInfo.pop("ty")]
```

Comment:
Aren't there any other types, e.g., `Scannable`?

Reply:
Oh, you are right!
There may be other types, but I planned not to handle them and assumed that handled experiments don't have those types.

Then, how about adding TODO comment?